micropython/umqtt.robust/umqtt/robust.py

```diff
@@ -42,3 +42,12 @@ def wait_msg(self):
             except OSError as e:
```

Comment:
Any failures in `super().wait_msg()` will be caught here, reconnected and looped (in blocking mode). As such it'll never get back up to the the `check_msg()` loop until this returns a valid response.

Reply:
Yes, and precisely the reason calling `**self**.wait_msg()` from my `check_msg()` override was a bad idea.

Reply:
Ah yes, you could quite likely ignore most of my suggestion below and instead go straight to super here instead of self!